### PR TITLE
[feat] Add the option to save a figure in plot setting params

### PR DESCRIPTION
**Pull request by nabenabe0928**
Since non-GUI based environments would like to avoid the usage of
show method in the matplotlib, I added the option to savefig and
thus users can complete the operations inside AutoPytorch.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: autoPyTorch/utils/results_visualizer.py [hunk 1]
```diff
@@ -71,8 +71,7 @@ def extract_dicts(
         return colors, labels
 
 
-@dataclass(frozen=True)
-class PlotSettingParams:
+class PlotSettingParams(NamedTuple):
```
Comment (nabenabe0928) [inline autoPyTorch/utils/results_visualizer.py:74]: This change is due to the fact that `dataclass` does not allow to have `dict` or `list`.

File: autoPyTorch/utils/results_visualizer.py [hunk 2]
```diff
@@ -201,17 +220,22 @@ def _set_plot_args(
...
+            ax.legend(**plot_setting_params.legend_kwargs)
 
         if plot_setting_params.title is not None:
-            ax.set_title(plot_setting_params.title)
-        if plot_setting_params.show:
+            ax.set_title(plot_setting_params.title, **plot_setting_params.title_kwargs)
+
+        if plot_setting_params.figname is not None:
+            plt.savefig(plot_setting_params.figname, **plot_setting_params.savefig_kwargs)
+        elif plot_setting_params.show:
```
Comment (ravinkohli) [inline autoPyTorch/utils/results_visualizer.py:238]: This means that show and figname would be mutually exclusive. Like we won't show the figure if there is a figure name even if `show==True`. Do you think we should do something like this:
```suggestion
        if plot_setting_params.show:
```

Comment (ravinkohli) [reply autoPyTorch/utils/results_visualizer.py:238]: ohh I see that is intended.

**Review by ravinkohli — APPROVED** [review]

**Comment by nabenabe0928** [comment]
[issue#350](https://github.com/automl/Auto-PyTorch/issues/350) was solved here.

**Pull request merged**